reference/cd-gitlab-ci.md

```diff
@@ -0,0 +1,183 @@
+
+## Scripts
+
+Your repository must contain the `.gitlab-ci.yml` in the root. GitLab looks there by default.
+If you are using an alternate location, be sure to update the settings for your GitLab project
+by going to https://gitlab.com > (select your project) > Settings > General.
+
+The following are samples only. You may choose to structure your configuration any way you like.
+The `setup.sh` installs the `pulumi` CLI on the GitLab CI Runner, and other tools.
+It also installs `yarn` and `nodejs` since that's the runtime for this sample project.
+
+The `run-pulumi.sh` script runs the `pulumi up` command to apply any stack changes and to start
+updating your infrastructure.
+
+The `pulumi-preview.sh` script (not shown here) is similar to the `run-pulumi.sh`, except that
+it runs the `pulumi preview` command instead of the `pulumi up` command, which is sort of a dry-run
+that only shows you changes (if any) in your infrastructure. Isn't that awesome?
+
+### Sample `.gitlab-ci.yml`
+```yaml
+#
+# This sample yaml configuration file contains 2 stages and 3 jobs.
+# This configuration uses GitLab's `only`, `when`, and `except` configuration
+# options to create a pipeline that will create the `pulumi-preview` job in the pipeline,
+# for all branches except the master.
+# Only for master branch merges, the main `pulumi` job is executed automatically.
+#  
+
+stages:
+  - build
+  - infrastructure-update
+
+# Each stage may require multiple jobs to complete that stage.
+# Consider a build stage, which may require building the UI, service, and a CLI.
+# All 3 individual build jobs can be attributed to the build _stage_.
+complex_build_job:
+  stage: build
+  script:
+    - echo "pulumi rocks!"
+
+# The 
+pulumi:
+  stage: infrastructure-update
+  before_script:
+    - chmod +x ./scripts/*.sh
+    - ./scripts/print_environment.sh
+    - ./scripts/setup.sh
+  script:
+    - ./scripts/run-pulumi.sh
+  # Create an artifact archive with just the pulumi log file,
+  # which is created using console-redirection in run-pulumi.sh.
+  artifacts:
+    paths:
+    - pulumi-log.txt
+    # This is just a sample of how artifacts can be expired (removed) automatically in GitLab.
+    # You may choose to not set this at all based on your organization's or team's preference.
+    expire_in: 1 week
+  # This job should only be created if the pipeline is created for the master branch.
+  only:
+    refs:
+      - master
+pulumi-preview:
+  stage: infrastructure-update
+  before_script:
+    - chmod +x ./scripts/*.sh
+    - ./scripts/print_environment.sh
+    - ./scripts/setup.sh
+  script:
+    - ./scripts/pulumi-preview.sh
+  # This job may only be triggered manually,
+  # even though the pipeline is created, 
+  # when a non-master branch is pushed-up.
+  when: manual
+  # This job should be created in the pipeline only if it is not the master branch.
+  except:
+    refs:
+      - master
+```
+
+### `setup.sh`
+
```

Comment:
Could use a quick intro for what this file is and does.  Same with `run-pulumi.sh` below. Even though you explain it earlier, the reader will likely skim to this section and have no context.